### DISCUSSION TITLE
Set a new attribute to Discord object

### DISCUSSION THREAD

**Discussion by CanavarB** [Rapptz/discord.py#9191] [Q&A] [answered]
How can I set an attribute to an object?

For example I want to add 'FORUM' constant to a discord.Guild object.

`self.GUILD  : discord.Guild = discord.utils.get(self.guilds, id = GUILD_ID) `
`self.GUILD.FORUM : discord.ForumChannel = discord.utils.get(self.GUILD.forums, id=FORUM_CHANNEL_ID)`

and also

`self.GUILD.FORUM.SPECIAL_TAG  : discord.ForumTag = discord.utils.get(self.GUILD.FORUM.available_tags, name='special name')`

edit: 
I tried setatrr() function too.

**Comment by LostLuma** [comment, marked as answer]
Hello, setting custom attributes on the libraries' types is not possible as they are [slotted](https://docs.python.org/3/reference/datamodel.html#slots) for a lower memory footprint.
You'll have to assign these attributes to the client / bot or another namespace within your code, similar to your `self.GUILD` attribute.
